x-pack/plugins/lens/public/drag_drop/drag_drop.tsx

```diff
@@ -248,3 +305,207 @@ const DragDropInner = React.memo(function DragDropInner(
     onDragStart: dragStart,
   });
 });
+
+const getKeyboardReorderMessageMoved = (position: number, prevPosition: number) =>
+  i18n.translate('xpack.lens.dragDrop.elementMoved', {
+    defaultMessage: 'You have moved the item from position {prevPosition} to position {position}',
+    values: {
+      position,
+      prevPosition,
+    },
+  });
+
+const getKeyboardReorderMessageLifted = (position: number) =>
+  i18n.translate('xpack.lens.dragDrop.elementLifted', {
+    defaultMessage: 'You have lifted an item in position {position}',
+    values: {
+      position,
+    },
+  });
+
+const lnsLayerPanelDimensionMargin = 4;
+
+export const ReorderableDragDrop = ({
+  draggingProps,
+  dropProps,
+  children,
+  label,
+  dropTo,
+}: {
+  draggingProps: {
+    className: string;
+    draggable: Props['draggable'];
+    onDragEnd: (e: DroppableEvent) => void;
+    onDragStart: (e: DroppableEvent) => void;
+    dataTestSubj: string;
+    isReorderDragging: boolean;
+  };
+  dropProps: {
+    onDrop: (e: DroppableEvent) => void;
+    onDragOver: (e: DroppableEvent) => void;
+    onDragLeave: () => void;
+    dragging: DragContextState['dragging'];
+    droppable: DraggableProps['droppable'];
+    itemsInGroup: string[];
+    id: string;
+    isActive: boolean;
+  };
+  children: React.ReactElement;
+  label: string;
+  dropTo: DropToHandler;
+}) => {
+  const { itemsInGroup, dragging, id, droppable } = dropProps;
+  const { reorderState, setReorderState } = useContext(ReorderContext);
+
+  const { isKeyboardReorderOn, reorderedItems, draggingHeight, direction, groupId } = reorderState;
+
+  const groupLength = itemsInGroup.length;
+
+  const currentIndex = itemsInGroup.indexOf(id);
+
+  const draggingClasses = classNames(
+    draggingProps.className,
+    {
+      'lnsDragDrop-isKeyboardModeActive': isKeyboardReorderOn,
+    },
+    {
+      'lnsDragDrop-isReorderable': draggingProps.isReorderDragging,
+    }
+  );
+
+  return (
+    <div className="lnsDragDrop__reorderableContainer">
+      <EuiScreenReaderOnly showOnFocus>
+        <button
+          aria-label={`Grab ${label}`}
```

Comment:
Do we need to add the text "Grab" in front of the label? It seems like this is not standard, EUI doesn't do it. Basically the screen reader is reading 1. The label itself then 2. The description saying that this item is draggable